lib/rouge/lexer.rb

```diff
@@ -501,12 +521,47 @@ def self.detect?(text)
   end
 
   module Lexers
+    ProxyLexer = Struct.new :path, :name, :tag, :aliases, :filenames, :mimetypes
+
+    @_loaded_files = {}
     @_loaded_lexers = {}
+    @_loaded_proxies = {}
+    @_source_files = []
+
+    def self.add_lexer(class_name)
+      return if @_source_files.empty?
+      @_loaded_files[class_name] = @_source_files.last
+    end
 
     def self.load_lexer(relpath)
       return if @_loaded_lexers.key?(relpath)
       @_loaded_lexers[relpath] = true
+      @_source_files.push relpath
       load File.join(__dir__, 'lexers', relpath)
+      @_source_files.pop
+    end
+
+    def self.load_proxies(hashes)
+      hashes.each do |h|
+        next if h["tag"].nil?
```

Comment:
Can't we filter out `false` values as well..?
```suggestion
        next unless h["tag"]
```

Reply:
We could but `tag` should never be false. It's either nil or a string.

Reply:
> but `h["tag"]` should never be false.

All the more reason why we should filter it out as well.

Reply:
I mean it's broken if `h["tag"]` is false and there should be some kind of error if that's the case. Filtering it out would cause it to fail silently.

Reply:
We really don't need to code defensively, since this data is generated by our own rake task and lives in our repository.

Reply:
Yeah, this was me writing hacky code. The correct place to put the check is in the Rake task that generates the JSON data (which is where it now is).